File: programs/onlybags/src/processors/update_state.rs
use anchor_lang::prelude::*;
use crate::instructions::update_state::UpdateState;

pub fn exec(
  ctx: Context<UpdateState>,
  staking_program_state: Pubkey,
  protocol_fee: u64,
  trade_fee_bps: u64,
  creator_fee: u64,
  total_token_supply: u64,
  staking_allocation: u64,
) -> Result<()> {
  let state = &mut ctx.accounts.state;
  state.staking_program_state = Some(staking_program_state);
  state.protocol_fee = protocol_fee;
  state.trade_fee_bps = trade_fee_bps;
  state.creator_fee = creator_fee;
  state.total_token_supply = total_token_supply;
  state.staking_allocation = staking_allocation;

  Ok(())
}

File: scripts/liquidos-curve/updateState.js
import * as anchor from "@coral-xyz/anchor";
import Web3Pkg from "@apocentre/solana-web3";
import {provider} from "../helpers/provider.js";
import {createAndSendV0Tx} from "../helpers/tx.js";
import config from "../config.v2.1.json" with { type: "json" };

const Web3 = Web3Pkg.default;
const {BN} = anchor.default;
const {PublicKey} = anchor.web3

const main = async () => {
  const state = new PublicKey(config.liquidosCurveState);
  const program = anchor.workspace.LiquidosCurve;
  const stakingProgram = anchor.workspace.LiquidosStaking;
  const deployer = provider.wallet.payer;
  const web3 = Web3(deployer.publicKey);

  const ix = await program.methods
  .updateState(
    stakingProgram.programId,
    new PublicKey(config.stakingState),
    new BN(config.protocolFee),
    new BN(config.tradeFeeBps),
    new BN(config.creatorFee),
    new BN(config.totalTokenSupply),
    new BN(config.stakingAllocation),
  )
  .accounts({
    state,
    owner: deployer.publicKey,
  })
  .instruction();

  const priorityFeeIx = web3.setComputeUnitPrice(20000);
  await createAndSendV0Tx(
    provider,
    [priorityFeeIx, ix],
    deployer.publicKey,
    [deployer]
  );

}

main()
.then(() => console.log("Success"))
.catch(error => console.log("Error: ", error))
